[Workers] readAtLeast revision (#2908)
cloudflare/cloudflare-docs: products/workers/src/content/runtime-apis/streams/readablestreambyobreader.md

  - Returns a promise with the next available chunk of data read into a passed-in buffer.

> - <Code>readAtLeast(size, buffer<ParamType>ArrayBufferView</ParamType>)</Code> <TypeLink href="https://streams.spec.whatwg.org/#dictdef-readablestreambyobreadresult">Promise&lt;ReadableStreamBYOBReadResult></TypeLink>

Reviewer: ```suggestion
- <Code>readAtLeast(minBytes, buffer<ParamType>ArrayBufferView</ParamType>)</Code> <TypeLink href="https://streams.spec.whatwg.org/#dictdef-readablestreambyobreadresult">Promise&lt;ReadableStreamBYOBReadResult></TypeLink>
```
For consistency with the comment below.

  `read` provides no control over the minimum number of bytes that should be read into the buffer. Even if you allocate a 1MiB buffer, the kernel is perfectly within its rights to fulfill this read with a single byte, whether or not an EOF immediately follows.

>   In practice, the Workers teams has found that `read` typically fills only 1% of the provided buffer.

Reviewer: ```suggestion
  In practice, the Workers team has found that `read` typically fills only 1% of the provided buffer.
```

>   The Workers team added `.readAtLeast()` to the Streams API to allow users to specify `minBytes` that should be read into the buffer before resolving the read.

Reviewer: What do you think about this?
```suggestion
  `readAtLeast` is a non-standard extension to the Streams API which allows users to specify that at least `minBytes` bytes must be read into the buffer before resolving the read.
```

  </Aside>

## Related resources

> - [Using Streams.](/learning/using-streams)
> - [Background about BYOB readers in the Streams API WHATWG specification.](https://streams.spec.whatwg.org/#byob-readers)

Author: ```suggestion
- [Using Streams](/learning/using-streams)
- [Background about BYOB readers in the Streams API WHATWG specification](https://streams.spec.whatwg.org/#byob-readers)
```
